pkg/util/pods.go

```diff
@@ -193,16 +189,11 @@ func waitForPodCompletionOrTimeout(ctx context.Context, podClient kubernetes.Pod
 				logrus.WithError(err).Warnf("Failed to get pod %s.", name)
 				continue
 			}
-
-			if !podSeenRunning {
-				if podHasStarted(pod) {
-					podSeenRunning = true
-				} else if time.Since(pod.CreationTimestamp.Time) > podStartTimeout {
-					message := fmt.Sprintf("pod didn't start running within %s: %s\n%s", podStartTimeout, getReasonsForUnreadyContainers(pod), getEventsForPod(ctx, pod, podClient))
-					logrus.Infof(message)
-					notifier.Complete(name)
-					return pod, errors.New(message)
-				}
+			if err := checkPending(*pod, pendingTimeout, time.Now()); err != nil {
+				err = fmt.Errorf("pod pending timeout: %w: %s\n%s", err, getReasonsForUnreadyContainers(pod), getEventsForPod(ctx, pod, podClient))
```

Comment:
So far, the pod (one of its containers) is stuck for whatever reason. What is going to happen next? Will it be eventually recollected by `k8s` somewhere in the future? If it won't, does it make sense to "manually" terminate it?

Reply:
What happens exactly depends on the nature of the original problem, but in all cases the pod will eventually be removed when the TTL of the temporary namespace expires.

Note that this PR changes nothing about how this is handled.

Reply:
> removed when the TTL of the temporary namespace expires

It looks fine.